source/components/temp_doeclim_component.cpp

```diff
@@ -235,7 +234,10 @@ void TempDOECLIMComponent::prepareToRun() throw ( h_exception ) {
     
     
     // Initializing all model components that depend on the number of timesteps (ns)
-    ns = core->sendMessage( M_GETDATA, D_END_DATE ) - core->sendMessage( M_GETDATA, D_START_DATE ) + 1;
+    ns = core->getEndDate() - core->getStartDate() + 1;
+    if( ns > 1000 ){
+        H_THROW("TempDOECLIM: Number of years larger than hard-coded max (1000). Change in temp_doeclim_component.hpp")
```

Comment:
This error is weak.  If the reason for the limit is something in _our_ code, then we need to just fix it using vectors that can be resized at run time.  

On the other hand, if the reason for the limit is fixed-size arrays in DOECLIM (or other code we can't change), then at a minimum we need to change the limit value to be a constant defined in a single place.  As it stands, if a user changes the value, they have to trawl through the code looking for all occurrences of the string "1000", change them too, and keep their fingers crossed that they got them all.  More importantly, if the reason for the limit is a limitation in DOECLIM, then trying to change it by upping the limit in Hector probably won't work, and if that's the case, then the error message should indicate as much.